backend/uclapi/common/decorators.py

```diff
@@ -0,0 +1,381 @@
+        return response
+
+    scopes = Scopes()
+    for s in required_scopes:
+        if not scopes.check_scope(token.scope.scope_number, s):
+            response = JsonResponse({
+                "ok": False,
+                "error":
+                    "The token provided does not have "
+                    "permission to access this data."
+            })
+            response.status_code = 400
+            return response
+
+    # Return the token as there are no issues
+    return token
+
+
+def _check_temp_token_issues(token_code, personal_data, request_path, page_token=None):
+    # The token is a generic one, so sanity check
+    if personal_data:
+        response = JsonResponse({
+            "ok": False,
+            "error": "Personal data requires OAuth."
+        })
+        response.status_code = 400
+        return response
+
+    try:
+        temp_token = TemporaryToken.objects.get(
+            api_token=token_code
+        )
+    except TemporaryToken.DoesNotExist:
+        response = JsonResponse({
+            "ok": False,
+            "error": "Invalid temporary token."
+        })
+        response.status_code = 400
+        return response
+
+    if request_path != "/roombookings/bookings":
+        response = JsonResponse({
+            "ok": False,
+            "error":
+                "Temporary token can only be used "
+                "for /bookings."
+        })
+        response.status_code = 400
+        return response
+
+    if page_token:
+        response = JsonResponse({
+            "ok": False,
+            "error":
+                "Temporary token can only return one booking."
+        })
+        response.status_code = 400
+        return response
+
+    # Check if TemporaryToken is still valid
+    existed = datetime.datetime.now() - temp_token.created
+
+    if existed.seconds > 300:
+        temp_token.delete()  # Delete expired token
+        response = JsonResponse({
+            "ok": False,
+            "error": "Temporary token expired."
+        })
+        response.status_code = 400
+        return response
+
+    # No issues, so return the temporary token
+    return temp_token
+
+
+def _check_general_token_issues(token_code, personal_data):
+    # The token is a generic one, so sanity check
+    if personal_data:
+        response = JsonResponse({
+            "ok": False,
+            "error": "Personal data requires OAuth."
+        })
+        response.status_code = 400
+        return response
+
+    try:
+        token = App.objects.get(
+            api_token=token_code,
+            deleted=False
+        )
+    except App.DoesNotExist:
+        response = JsonResponse({
+            "ok": False,
+            "error": "Token does not exist."
+        })
+        response.status_code = 400
+        return response
+
+    # No issues, so return the token
+    return token
+
+
+def uclapi_protected_endpoint(personal_data=False, required_scopes=[]):

```

Comment:
I would be wary using mutable variables as arguments. (and linter should be complaining too)

Reply:
What's the solution to this in this case? Remember that `required_scopes` is never written to by the decorator, only read from.